packages/gatsby-source-contentful/src/source-nodes.js

```diff
@@ -398,11 +376,33 @@ export async function sourceNodes(
         })
       }
     })
+  }
+
+  // allow node to gc if it needs to
+  // @ts-ignore
+  newOrUpdatedEntries = undefined
+  await new Promise(res => {
```

Comment:
This nasty promise triggers GC?

Reply:
There's no way to trigger GC but when the event loop is blocked v8 seems to not be able to GC. It's supposed to be able to but my guess is when blocking there isn't a safe way for it to know it can GC these kinds of objects.
These promises that resolve on the next event loop tick prevents this code from blocking

Reply:
I added a little util to make this clearer, lmk what you think!

Reply:
yeah thats great!